dissect/target/plugins/os/windows/regf/mru.py

```diff
@@ -353,30 +362,63 @@ def parse_mru_key(target, key, record):
 def parse_mru_ex_key(target, key, record):
     user = target.registry.get_user(key)
 
-    mrulist_ex = key.value("MRUListEx").value
-    mrulist_ex = struct.unpack(f"<{len(mrulist_ex) // 4}I", mrulist_ex)
+    try:
+        mrulist_ex = key.value("MRUListEx").value
+        mrulist_ex = struct.unpack(f"<{len(mrulist_ex) // 4}I", mrulist_ex)
+    except RegistryError:
+        mrulist_ex = None
 
     for value in key.values():
         if value.name == "MRUListEx":
             continue
 
         entry_index = mrulist_ex.index(int(value.name))
-        split_idx = value.value.index(b"\x00\x00")
-        # Poor mans null terminated utf-16-le
-        path, bag = value.value[: split_idx + 1], value.value[split_idx + 3 :]
+        if record != OpenSaveMRURecord:
+            split_idx = value.value.index(b"\x00\x00")
+            # Poor mans null terminated utf-16-le
+            path, bag = value.value[: split_idx + 1], value.value[split_idx + 3 :]
+        else:
+            bag = value.value
+            path = None
         parsed_bag = list(parse_shell_item_list(bag))
         if len(parsed_bag) != 1 or not isinstance(parsed_bag, FILE_ENTRY):
             target.log.debug("Unexpected shell bag entry in MRUListEx entry: %s:%s", key, value)
+        filepath = get_filepath_from_bag(parsed_bag)
+
+        if record == LastVisitedMRURecord:
+            yield record(
+                regf_mtime=key.ts,
+                index=entry_index,
+                filename=path.decode("utf-16-le"),
+                path=filepath,
+                key=key.path,
+                _target=target,
+                _user=user,
+                _key=key,
+            )
+        elif record == OpenSaveMRURecord:
+            yield record(
+                regf_mtime=key.ts,
+                index=entry_index,
+                value=filepath.rstrip("\\"),
+                key=key.path,
+                _target=target,
+                _user=user,
+                _key=key,
+            )
+        else:
+            yield record(
+                regf_mtime=key.ts,
+                index=entry_index,
+                value=path.decode("utf-16-le"),
+                key=key.path,
```

Comment:
Just a general comment, I think we can get rid of the `key` value in this record, since the registry record extension will add it as `regf_key_path` already.